rp2040-hal/src/timer.rs

```diff
@@ -268,7 +268,7 @@ macro_rules! impl_alarm {
                 if duration < MIN_MICROSECONDS {
                     return Err(ScheduleAlarmError::AlarmTooSoon);
                 } else {
-                    cortex_m::interrupt::free(|_| {
+                    critical_section::with(|_| {
```

Comment:
I don't think this is a good change. The only reason interrupts are disabled here is that the timing is critical. It's not done to avoid concurrent accesses.
Changing this to a critical section only increases the overhead.
(In general, a global critical sections and disabling interrupts can serve different purposes. It may have been a mistake to return a CriticalSection token in both cases. But on single core, it's the same operation, so it's understandable.)